lib/node_modules/@stdlib/complex/parse-float64/lib/main.js

```diff
@@ -0,0 +1,76 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var Complex128 = require( '@stdlib/complex/float64' );
+
+
+// MAIN //
+
+/**
+* Parse a string representation of a complex number and returns a Complex128 instance.
+*
+* @param {string} str - string representation of a complex number
+* @throws {TypeError} must provide a string
+* @throws {Error} must provide a valid string representation of a complex number
+* @returns {Complex128} 128-bit complex number
+*
+* @example
+* var str = '1 + 2j';
+*
+* var z = parseComplex128( str );
+* // returns <Complex128>
+*/
+function parseComplex128(str) {
+	var regex;
+	var match;
+	var re;
+	var im;
+
+	if (typeof str !== 'string') {
+		throw new TypeError('Invalid input: expected a string.');
+	}
+
+	// Regular expression to match a complex number in the form 'a + bj', 'a - bj', 'bj', or 'a'
+	regex = /^([-+]?(\d*\.?\d*(?:[eE][-+]?\d+)?|Infinity|NaN)j?)?([-+])?((\d*\.?\d*(?:[eE][-+]?\d+)?|Infinity|NaN)j)?$/;
```

Comment:
For an example of how we typically document regular expressions, see https://github.com/stdlib-js/stdlib/blob/952ea72db18b6ff6b06bf89a3559d9f99a69cae5/lib/node_modules/%40stdlib/regexp/regexp/lib/regexp.js#L29.

Better yet, you can move this regexp to a helper function and then use doctesting to confirm that it works as expected.

L54 would then become

```
match = str.replace( /\s/g, '' ).match( regexp() );
```

Also, instead of string prototype methods, prefer their functional counterparts. In this case, see `@stdlib/string/base/replace`.